### ISSUE TITLE
Possibly unintended change in layeredimage conflict managing

### ISSUE THREAD

**Issue by Gouvernathor** [renpy/renpy#2528]
040ec21 changed stuff in the choice of which displayable to choose, when one or several of those are layeredimage or live2D.
In renpy/display/image.py/choose_image(), displayables are polled and the ones with the most attributes, when they include all required ones, are stored in `matches`.
However, the attributes already being in a layeredimage's (or live2D's) name ("sitting" if it was declared as `layeredimage eileen sitting`) are counted, and do not bar it from being part of `matches`, when it would be the case were it not a layeredimage (nor a live2D).
This happens _before_ my last commit, which only matters when several matches are found with the same number of attributes.
I'll show an example where it has an influence on the final result.
```
layeredimage eileen one:
    size (400, 400)
    xalign .0 # shows on the left
    attribute aa default:
        Solid('#c60')
    attribute ab:
        Solid('#f00')
    attribute base:
        Solid('#0f0')

layeredimage eileen:
    size (400, 400)
    xalign 1.0 # shows on the right
    attribute ba default:
        Solid('#60c')
    attribute bb:
        Solid('#00f')
    attribute base:
        Solid('#0f0')
```
Say I call `show eileen base`. The polling finds `('one', 'base')` for eileen one, and `('base',)` for eileen.
With the old behavior, 'one' being neither required nor optional, `eileen one` would be barred from the `matches` list, leaving `eileen` as the chosen one. (a green square would appear on the right)
With the new behavior, since `eileen one` is a layeredimage, the fact that it has a non-optional attribute is waived, and it is kept in the tally with 2 attributes. `eileen` doesn't make it, having only 1. `eileen one` is chosen and displayed. (the green square appears on the left)

TL;DR now when several poses for the same layeredimage can respond to the requested attributes, the ones with the largest amount of attributes in their name are given an unfair advantage, whereas they used to be banned.

You might think it's stupid to have an eileen with a pose and eileen with none, and it's probably true, but the same thing would be happening would we have an `eileen one`, an `eileen two sitting` and `eileen two standing`.

**Comment by Gouvernathor** [comment]
To get to the solving section, maybe if they were only tallied regarding their amount of optional attributes in the first place, instead of their total amount, it would solve the issue. While still requiring all required attributes to be there of course.
it would mean replacing line 985 :
```
len_attrs = len(set(attrs))
```
with :
```
len_attrs = len(set(attrs).intersection(optional)) # number of optionals in this match
```
And then discarding the useless #2524 second tally.

I can't see how it would
